uniffi_bindgen/src/bindings/swift/templates/RecordTemplate.swift

```diff
@@ -1,4 +1,4 @@
-public struct {{ rec.name()|class_name_swift }}:  ViaFfiUsingByteBuffer, ViaFfi, Equatable, Hashable {
+public struct {{ rec.name()|class_name_swift }}: Equatable, Hashable {
```

Comment:
The general pattern here is: 

1. Make the uniffi protocol `fileprivate`, so we can't see it outside of this file.
2. Split the implementing `extension` off from the public extensions/declaration. This is optional, but is more swifty.
3. Implement the `fileprivate` protocol in a vanilla extension, but make all members `fileprivate`.

Neither the methods nor the protocols should now be visible outside this file.

Reply:
This is going to lightly interfere with what I was trying to do over in https://github.com/mozilla/application-services/pull/3900, but since that PR is just a hypothetical experiment, I don't think we should change things in UniFFI to support it. Your proposal here sounds good.